ee/maintained-apps/inputs/darwin/ingester.go

```diff
@@ -0,0 +1,370 @@
+		return nil, nil, ctxerr.Wrap(ctx, err, "execute http request")
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, nil, ctxerr.Wrap(ctx, err, "read http response body")
+	}
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		// success, go on
+	case http.StatusNotFound:
+		// nothing to do
+		level.Warn(i.logger).Log("msg", "maintained app missing in brew API", "identifier", app.SourceIdentifier)
+		return nil, nil, nil
+	default:
+		if len(body) > 512 {
+			body = body[:512]
+		}
+		return nil, nil, ctxerr.Errorf(ctx, "brew API returned status %d: %s", res.StatusCode, string(body))
+	}
+
+	var cask brewCask
+	if err := json.Unmarshal(body, &cask); err != nil {
+		return nil, nil, ctxerr.Wrapf(ctx, err, "unmarshal brew cask for %s", app.SourceIdentifier)
+	}
+
+	out := &maintained_apps.FMAManifestApp{}
+
+	// validate required fields
+	if len(cask.Name) == 0 || cask.Name[0] == "" {
+		return nil, nil, ctxerr.Errorf(ctx, "missing name for cask %s", app.SourceIdentifier)
+	}
+	if cask.Token == "" {
+		return nil, nil, ctxerr.Errorf(ctx, "missing token for cask %s", app.SourceIdentifier)
+	}
+	if cask.Version == "" {
+		return nil, nil, ctxerr.Errorf(ctx, "missing version for cask %s", app.SourceIdentifier)
+	}
+	if cask.URL == "" {
+		return nil, nil, ctxerr.Errorf(ctx, "missing URL for cask %s", app.SourceIdentifier)
+	}
+	_, err = url.Parse(cask.URL)
+	if err != nil {
+		return nil, nil, ctxerr.Wrapf(ctx, err, "parse URL for cask %s", app.SourceIdentifier)
+	}
+
+	out.Version = cask.Version
+	out.InstallerURL = cask.URL
+	out.UniqueIdentifier = app.UniqueIdentifier
+	out.SHA256 = cask.SHA256
+	out.Queries = map[string]string{maintained_apps.QueryKeyExists: fmt.Sprintf("SELECT 1 FROM apps WHERE bundle_identifier = '%s';", out.UniqueIdentifier)}
+	out.Description = cask.Desc
+	out.Slug = fmt.Sprintf("%s/%s", cask.Token, fleet.MacOSPlatform)
+
+	// Script generation
+	scriptRefs := make(map[string]string)
+	uninstallScript := uninstallScriptForApp(&cask)
+	uninstallRef := uuid.NewString()
+	out.UninstallScriptRef = uninstallRef
+	installScript, err := installScriptForApp(app, &cask)
+	if err != nil {
+		return nil, nil, ctxerr.WrapWithData(ctx, err, "generating install script for maintained app", map[string]any{"unique_identifier": app.UniqueIdentifier})
+	}
+	installRef := uuid.NewString()
+	scriptRefs[installRef] = installScript
+	out.InstallScriptRef = installRef
+
+	scriptRefs[uninstallRef] = uninstallScript
+
+	return out, scriptRefs, nil
+}
+
+func shouldUpdateApp(ctx context.Context, sourceID string, outApp *maintained_apps.FMAManifestApp) (bool, error) {
+	existingFileBytes, err := os.ReadFile(path.Join(outputPath, string(fleet.MacOSPlatform), fmt.Sprintf("%s.json", sourceID)))
+	if err != nil {
+		if err == os.ErrNotExist {
+			return true, nil
+		}
+
+		return false, ctxerr.Wrap(ctx, err, "reading existing app manifest")
+	}
+
+	var existingManifest maintained_apps.FMAManifestFile
```

Comment:
Seems like for parsing errors, or no versions, we should return true here, so if an improperly formatted file gets committed we overwrite it.